src/Microsoft.Azure.SignalR.Common/Endpoints/IConfigurationExtension.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.SignalR.Common.Endpoints
+{
+    public static class IConfigurationExtension
+    {
+        public static (string ConnectionString, ServiceEndpoint[]) GetSignalRServiceEndpoints(this IConfiguration configuration, string sectionKey)
```

Comment:
what does the return value mean? one connectionstring but mulitple endpoints? is this `connectionString` another `service endpoint`? Is it a valid case that both have value?

Reply:
`ConnectionString` is the value with key `Azure:SignalR:ConnectionString`, `ServiceEndpoint[]` are endpoints with key `Azure:SignalR:ConnectionString:...` whose name or `ServiceEndpointType` is explicitly defined.

Reply:
so connectionString is another ServiceEndpoint with some default properties?

Reply:
This is a interface from original [`ServiceOptionsSetup`](https://github.com/Azure/azure-signalr/blob/0ff0e066af962927f84ef8e86aa045a0f6acbcee/src/Microsoft.Azure.SignalR/ServiceOptionsSetup.cs#L60), following this interface makes things easier. I think why sperate `Azure:SignalR:ConnectionString` and others service endpoint is that `Azure:SignalR:ConnectionString` is on the higher configuration lever than other keys, and they are parse in different ways.